### PR TITLE
bip32: Add increment method to DerivationPath and ChildNumber

### PR DESCRIPTION
**Pull request by stevenroose**

### CODE REVIEW

File: src/util/bip32.rs [hunk 1]
```diff
@@ -261,19 +268,36 @@ impl FromStr for DerivationPath {
...
-    /// Convert into a DerivationPath that is a child of this one.
+    /// Convert into a [DerivationPath] that is a child of this one.
     pub fn into_child(self, cn: ChildNumber) -> DerivationPath {
         let mut path = self.0;
         path.push(cn);
         DerivationPath(path)
     }
+
+    /// Returns the [DerivationPath] that is a single increment from this one.
+    pub fn increment(&self) -> Result<DerivationPath, Error> {
```
Comment (sgeisler) [inline src/util/bip32.rs]: Is the increment of a path well-defined? I couldn't find it in the standard and just incrementing the last part seems a bit arbitrary (although I see the use case). Maybe a child iterator that starts at a certain child number and yields all following children would solve the problem too with the added benefit of access to the iterator API? Then a wallet could do the following to calculate its balance:

```rust
let first_normal = ChildNumber::from_normal_idx(0).unwrap();
let path: DerivationPath = "m/0/0'".parse().unwrap();

let balance = path.children_since(first_normal)
    .flat_map(get_utxos_for_path)
    .map(utxo_amount)
    .sum();
```

Comment (apoelstra) [reply src/util/bip32.rs]: +1 to having an iterator instead

Comment (stevenroose) [reply src/util/bip32.rs]: Yeah that makes sense. Take a look at the new push.

**Comment by apoelstra** [comment]
Looks good! But can you remove the `increment` commit? :)

**Comment by stevenroose** [comment]
Sure I can squash them.

**Comment by stevenroose** [comment]
Squashed.

**Comment by apoelstra** [comment]
For the record, I had meant to actually remove the `increment` function rather than squashing the commits together. But ok, it's not like it's hurting anything :)

**Pull request merged**